File: 2022/day01b.rs
use std::io::{BufReader, BufRead};
use std::fs::File;
// use std::str::FromStr;

fn main() { 
    let mut calories_per_elf: Vec<u32> = Vec::new();
    let f = File::open("day01.txt").unwrap();
    let reader = BufReader::new(f);
    let mut total = 0;
    for line in reader.lines() {
        let line = line.unwrap();
        match line.trim().parse::<u32>() {
            Ok(n) => total += n,
            Err(_) => { calories_per_elf.push(total); total = 0 }
        };
    }
    calories_per_elf.sort();
    let top_3_elves:u32 = calories_per_elf.iter().rev().take(3).sum();
    println!("The sum of the calories carried by the top 3 elves is {} calories.", top_3_elves);
}

File: 2022/day01b.py
#!/usr/bin/env python3

if __name__ == "__main__":
    with open("day01.txt", "r") as f:
        data = f.read().splitlines()
    calories_per_elf = []
    running_total = 0
    for line in data:
        if not len(line):
            calories_per_elf.append(running_total)
            running_total = 0
        else:
            running_total += int(line)
    print(f"The elf with the most calories has {max(calories_per_elf)} calories.")
    print(f"The sum of the calories carried by the top 3 elves is {sum(sorted(calories_per_elf)[-3:])} calories.")

